cmd/branchingconfigmanagers/generated-release-gating-jobs/main.go

```diff
@@ -0,0 +1,155 @@
+	howManyDaysBefore                  = 14
+	lifecycleConfigFilePathDefaultPath = ""
+)
+
+type options struct {
+	curOCPVersion           string
+	futureOCPVersion        string
+	releaseJobsDir          string
+	releaseRepoDir          string
+	outDir                  string
+	lifecycleConfigFilePath string
+	logLevel                int
+	newIntervalValue        int
+}
+
+func (o *options) overrideOCPVersions(curVersion, futureVersion string) {
+	if o.curOCPVersion == "" {
+		o.curOCPVersion = curVersion
+	}
+	if o.futureOCPVersion == "" {
+		o.futureOCPVersion = futureVersion
+	}
+}
+
+func (o *options) adjustPaths() error {
+	if o.releaseRepoDir != "" {
+		if !path.IsAbs(o.releaseRepoDir) {
+			return errors.New("release-repo-path path has to be absolute")
+		}
+
+		if path.IsAbs(o.releaseJobsDir) {
+			return errors.New("release-jobs path can't be absolute when release-jobs is specified")
+		}
+
+		o.releaseJobsDir = path.Join(o.releaseRepoDir, o.releaseJobsDir)
+	}
+	return nil
+}
+
+func gatherOptions() (*options, error) {
+	var errs []error
+	o := &options{}
+	flag.StringVar(&o.curOCPVersion, "current-release", "", "Current OCP version")
+	flag.StringVar(&o.futureOCPVersion, "future-release", "", "Next OCP version")
+	flag.StringVar(&o.releaseJobsDir, "release-jobs", releaseJobsPath, "Path to release-gating jobs .yaml files")
+	flag.StringVar(&o.releaseRepoDir, "release-repo-path", "", "Path to 'openshift/release/ folder")
+	flag.StringVar(&o.outDir, "out", "", "Output directory. Default to --release-jobs dir")
+	flag.StringVar(&o.lifecycleConfigFilePath, "lifecycle-config", lifecycleConfigFilePathDefaultPath, "Path to the lifecycle config file")
+	flag.IntVar(&o.newIntervalValue, "interval", 168, "New interval to set")
+	flag.IntVar(&o.logLevel, "log-level", int(logrus.DebugLevel), "Log level")
+	flag.Parse()
+
+	if _, err := ocplifecycle.ParseMajorMinor(o.curOCPVersion); o.curOCPVersion != "" && err != nil {
+		errs = append(errs, fmt.Errorf("error parsing current-release %s", o.curOCPVersion))
+	}
+
+	if _, err := ocplifecycle.ParseMajorMinor(o.futureOCPVersion); o.futureOCPVersion != "" && err != nil {
+		errs = append(errs, fmt.Errorf("error parsing future-release %s", o.futureOCPVersion))
+	}
+
+	if o.newIntervalValue < 0 {
+		errs = append(errs, errors.New("error parsing interval: value is not a positive integer"))
+	}
+
+	if o.lifecycleConfigFilePath == "" {
+		errs = append(errs, errors.New("lifecycle-config is mandatory"))
+	}
+	if !path.IsAbs(o.lifecycleConfigFilePath) {
+		errs = append(errs, errors.New("lifecycle-config path is not absolute"))
+	}
+
+	if o.outDir != "" && !path.IsAbs(o.outDir) {
+		errs = append(errs, errors.New("out path is not absolute"))
+	}
+
+	return o, utilerrors.NewAggregate(errs)
+}
+
+func main() {
+	o, err := gatherOptions()
+	if err != nil {
+		logrus.WithError(err).Error("failed to gather options")
+		os.Exit(optsParsingErrorExitStatus)
```

Comment:
```suggestion
		logrus.WithError(err).Fatal("failed to gather options")
```

You can remove the `os.Exit`